data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/cnec/FlowCnecImpl.java

```diff
@@ -0,0 +1,262 @@
+import com.farao_community.farao.commons.Unit;
+import com.farao_community.farao.data.crac_api.NetworkElement;
+import com.farao_community.farao.data.crac_api.Side;
+import com.farao_community.farao.data.crac_api.State;
+import com.farao_community.farao.data.crac_api.cnec.BranchCnec;
+import com.farao_community.farao.data.crac_api.threshold.BranchThreshold;
+import com.farao_community.farao.data.crac_api.threshold.Threshold;
+import com.farao_community.farao.data.crac_impl.NotSynchronizedException;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.CurrentLimits;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.TieLine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.farao_community.farao.data.crac_api.Side.LEFT;
+import static com.farao_community.farao.data.crac_api.Side.RIGHT;
+import static java.lang.String.format;
+
+/**
+ * @author Joris Mancini {@literal <joris.mancini at rte-france.com>}
+ */
+@JsonTypeName("flow-cnec")
+public class FlowCnecImpl extends AbstractBranchCnec implements BranchCnec {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowCnecImpl.class);
+    private static final String TIE_LINE_WARN = "For tie-line {}, the network element ID {} is not half1 nor half2 IDs. Most limiting threshold will be taken.";
+    private static final String IMAX_CONVERSION_WARN = "Imax on side {} for the network element {} will be converted.";
+
+    private final Double[] iMax = new Double[2];
+    private double frm;
+
+    public FlowCnecImpl(String id, String name, NetworkElement networkElement, State state, boolean optimized, boolean monitored, Set<BranchThreshold> thresholds, double frm) {
+        super(id, name, networkElement, state, optimized, monitored, thresholds);
+        this.frm = frm;
+    }
+
+    public FlowCnecImpl(String id, NetworkElement networkElement, State state, boolean optimized, boolean monitored, Set<BranchThreshold> thresholds, double frm) {
+        super(id, networkElement, state, optimized, monitored, thresholds);
+        this.frm = frm;
+    }
+
+    private Double getIMax(Side side) {
+        checkSynchronized(format("access iMax values of flow cnec %s", getId()));
+        if (side.equals(LEFT)) {
+            return iMax[0];
+        } else {
+            return iMax[1];
+        }
+    }
+
+    private void setIMax(Side side, Double value) {
+        if (side.equals(LEFT)) {
+            iMax[0] = value;
+        } else {
+            iMax[1] = value;
+        }
+    }
+
+    @Override
+    public Optional<Double> getLowerBound(Side side, Unit requestedUnit) {
+        if (side == null) {
+            throw new NotSynchronizedException(format("Flow cnec %s has to be synchronized with a network to be able to set a side.", getId()));
```

Comment:
I don't get the error message. Why not just "side cannot be null" or something similar ?

Reply:
The side can only be set by synchronization, ths is why. But it still precises that the error occured because a side is missing. Do you want me to change ?

Reply:
I don't get it, here the side is given in argument of the function. It is not an internal parameter of the FlowCnecImpl set during the synchronisation. So if it is null it means that the cient did `getLowerBound(null, Unit.MEGAWATT)` ? Or do I miss something ?